encoding/encoding.go

```diff
@@ -48,6 +48,14 @@ type Compressor interface {
 	Name() string
 }
 
+// CompressorSizer is optional, can be implemented to improve efficiency.
+// This API is EXPERIMENTAL.
+type CompressorSizer interface {
```

Comment:
I don’t understand how it could be optional that way.

Reply:
It would only appear in the comments.  I'm not suggesting adding `DecompressedSize` to `Compressor` itself.

Reply:
I.e. the type assertion in recvAndDecompress is:
```go
if ds, ok := decompressor.(interface { DecompressedSize([]byte) int32 }); ok {
  size = ds.DecompressedSize(...)
...
```

Reply:
OK, I see what you mean.
I don't feel strongly, but that seems worse to me.  For instance a named interface allows implementers to assert that their type conforms.

Reply:
This is fairly standard for Go.  E.g. in the [error wrapping](https://godoc.org/errors#Unwrap) design, the `errors.Wrapper` interface was ultimately removed because it wasn't necessary (and also nobody could agree on what to call it).